pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/mail/MailTemplateSubstitution.java

```diff
@@ -0,0 +1,19 @@
+package org.broadinstitute.ddp.model.mail;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Value
+@AllArgsConstructor
+public class MailTemplateSubstitution {
+    Map<String, String> values = new HashMap<>();
+
+    public MailTemplateSubstitution withValue(final String name, final String value) {
+        values.put(name, value);
+
+        return this;
+    }
```

Comment:
`HashMap` allows for keys to have an explicit `null` value. This will lead to weird NPEs down the line when applying the template and should be guarded against (this relates to my comment in [MailTemplateRepeatableElement.java](https://github.com/broadinstitute/ddp-study-server/pull/2168/files#diff-5660ee13d3903b43839ba2454eec5082e0b0ce8a015b7c2177b9926dfabd207d))

Reply:
Added not null check